### ISSUE TITLE
IDE can't find import in KMP module's commonMain source set

### ISSUE THREAD

**Issue by edenman** [cashapp/exhaustive#6]
I've added the plugin to both my android app module and my KMP module but the import for the `Exhaustive` annotation can't be found in the KMP module's `commonMain` source set.  It works fine in the `androidMain` source set.

https://github.com/edenman/kmpPlayground/compare/edenman/exhaustive-when

**Comment by JakeWharton** [comment]
What task are you running? Doing `gw :app:build` is successful. Doing a full `build` fails in yarn about an invalid module name.

**Comment by edenman** [comment]
Hm, I just meant AndroidStudio can't resolve the import in `FooProvider`

**Comment by edenman** [comment]
I've never actually built this thing for js/iOS, it's just sort of a stub project to show off stuff that is weird about KMP

**Comment by edenman** [comment]
I wonder if "The @Exhaustive annotation will be made available in your main and test source sets" means that it just finds the `androidMain` source set and doesn't add the dep to `commonMain`?

**Comment by JakeWharton** [comment]
I think it's an IDE plugin bug.

The classpaths all appear correct with the `dependencies` task and as I said a normal compile works.

If you add the annotation dependency manually to `commonMain` as `compileOnly("app.cash.exhaustive:exhaustive-annotation:0.1.1")` it will resolve. The Gradle plugin does this automatically, which is why we can see it in the `dependencies` task, but it seems like the IDE doesn't see that for some reason.

I created https://youtrack.jetbrains.com/issue/KT-43385 to track. Closing this unless they come back with changes needed on our side.
